Gems/Atom/RHI/Code/Include/Atom/RHI/DeviceFence.h

```diff
@@ -25,7 +25,7 @@ namespace AZ::RHI
         virtual ~DeviceFence() = 0;
 
         /// Initializes the fence using the provided device and initial state.
```

Comment:
Could you add the description of the new parameter to the comment.
Also, not sure if `usedForWaiting` is the best name, maybe `usedForWaitingOnGPU` or something along those lines to be more specific?